pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -196,6 +202,16 @@ def restore_loops(self) -> None:
         self.trainer.fit_loop.global_step = self._loaded_checkpoint["global_step"]
         self.trainer.fit_loop.current_epoch = self._loaded_checkpoint["epoch"]
 
+        assert self.trainer.state.fn is not None
+        state_dict = self._loaded_checkpoint.get("loops")
+        if state_dict is not None and self.trainer.state.fn != TrainerFn.TUNING:
+            loop_name = f"{self.trainer.state.fn}_loop"
+            loop_attr: Loop = getattr(self.trainer, loop_name)
```

Comment:
what happens if `loop_attr` isn't set on the trainer? the exception will be thrown in L210 when `None` tries to load the state dict. i'm with @carmocca that this naming scheme is not as enforced and can break with renames

Reply:
ok.. just trying to avoid those annoying if-else statements here.. I can do a simple dict mapping
```py
if state_dict is not None and self.trainer.state.fn != TrainerFn.TUNING:
    loop_attr_map = {TrainerFn.FITTING: trainer.fit_loop, TrainerFn.VALIDATING: trainer.validate_loop, ...}
    loop_attr: Loop = loop_attr_map[self.trainer.state.fn]
    loop_attr.load_state_dict(state_dict[loop_attr.__name__])
```

or just revert back to the if-else statements? @ananthsub ?

Reply:
i like the explicit if-else statements because then github/editors can surface the references, which helps in traceability and refactoring.